rclpy/rclpy/node.py

```diff
@@ -941,7 +950,10 @@ def _apply_descriptor(
 
         if descriptor.dynamic_typing:
             descriptor.type = parameter.type_.value
-        elif descriptor.type != parameter.type_.value:
+        elif (
+            parameter.type_ != Parameter.Type.NOT_SET and
```

Comment:
IIUC, this is allowing to set an statically typed parameter to PARAMETER_NOT_SET after it was declared.
That shouldn't be valid, a test case is also a good idea.

Reply:
You're right. I'm finding it tricky to handle this case with the current way the code is structured.

Reply:
This seems to do the trick: 5e8266f

Though, it seems a bit hacky to me to check if the parameter exists inside the "apply_descriptor" method.

Reply:
yeah, the logic is a bit cumbersome, but I think it's doing the correct thing :+1: